### ISSUE TITLE
k8s_external returns all services with ClusterIP on zone transfer

### ISSUE THREAD

**Issue by aleksey-novikov** [coredns/coredns#4801]
**What happened**: On zone transfer Coredns returns all contents of kubernetes plugin zone with domain suffix replaced by k8s_external plugin zone.
```
$ kubectl -n istio-system get svc
NAME                   TYPE           CLUSTER-IP       EXTERNAL-IP     PORT(S)                                                                      AGE
istio-egressgateway    ClusterIP      10.239.105.210   <none>          80/TCP,443/TCP                                                               15d
istio-ingressgateway   LoadBalancer   10.239.105.85    10.239.112.16   15021:32212/TCP,80:31201/TCP,443:32190/TCP,31400:30464/TCP,15443:32044/TCP   15d
istiod                 ClusterIP      10.239.105.123   <none>          15010/TCP,15012/TCP,443/TCP,15014/TCP                                        15d
kiali                  ClusterIP      10.239.104.84    <none>          20001/TCP,9090/TCP                                                           15d

```
```
$ dig AXFR @10.239.96.10 example.com. | grep 'istio-system.*IN[[:space:]]A'
istio-egressgateway.istio-system.svc.example.com. 5 IN A 10.239.105.210
istio-ingressgateway.istio-system.svc.example.com. 5 IN	A 10.239.105.85
istiod.istio-system.svc.example.com. 5 IN A	10.239.105.123
kiali.istio-system.svc.example.com. 5 IN A	10.239.104.84

```

**What you expected to happen**: On zone transfer k8s_external plugin should return records for serviceType=LoadBalancer only and its external IPs.

```
$ dig AXFR @10.239.96.10 example.com. | grep 'istio-system.*IN[[:space:]]A'
istio-ingressgateway.istio-system.svc.example.com. 5 IN	A 10.239.112.16

```
**How to reproduce it (as minimally and precisely as possible)**: Sample CoreDNS config
```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns
  namespace: kube-system
data:
  Corefile: |
    .:53 {
        errors
        health {
           lameduck 5s
        }
        ready
        kubernetes cluster.local in-addr.arpa ip6.arpa {
           pods insecure
           fallthrough in-addr.arpa ip6.arpa
           ttl 30
        }
        k8s_external example.com {
           ttl 30
        }
        transfer example.com {
          to *
        }
        prometheus :9153
        forward . /etc/resolv.conf {
           max_concurrent 1000
        }
        cache 30
        loop
        reload
        loadbalance
    }

```

**Environment**:
CoreDNS version: 1.8.4
Kubernetes version: 1.20.5
OS: irrelevant (CentOS 7/8/etc)

**Comment by chrisohaver** [comment]
> What you expected to happen: On zone transfer k8s_external plugin should return records for serviceType=LoadBalancer only and its external IPs.

Thanks!  _k8s_external_ doesn't actually implement zone transfer, so there should not be an answer at all.  This looks like a bug that is causing the wrong plugin to answer the transfer request (due to _kubernetes_ being in the same server block).   

I'm testing a fix now.

**Comment by chrisohaver** [comment]
Closing this since the bug here is fixed in #4802.

I have opened #4803 for the enhancement request of adding zone transfer for the _k8s_external_ plugin.
